libcaf_net/src/net/web_socket/framing.cpp

```diff
@@ -186,27 +181,34 @@ bool framing::end_text_message() {
 
 // -- implementation details ---------------------------------------------------
 
-bool framing::handle(uint8_t opcode, byte_span payload) {
-  // Code rfc6455::connection_close must be treated separately.
-  CAF_ASSERT(opcode != detail::rfc6455::connection_close);
+ptrdiff_t framing::handle(uint8_t opcode, byte_span payload,
+                          size_t frame_size) {
   switch (opcode) {
+    case detail::rfc6455::connection_close:
+      abort_and_shutdown(sec::connection_closed);
+      return -1;
     case detail::rfc6455::text_frame: {
       std::string_view text{reinterpret_cast<const char*>(payload.data()),
                             payload.size()};
-      return up_->consume_text(text) >= 0;
+      if (up_->consume_text(text) < 0)
+        return -1;
+      break;
     }
     case detail::rfc6455::binary_frame:
-      return up_->consume_binary(payload) >= 0;
+      if (up_->consume_binary(payload) < 0)
```

Comment:
@Neverlord  Should we use `<0` or `<1` here?

Reply:
We don't want to change the logic, so `< 0`.